app/src/main/java/com/example/jwplayerdemo/JWMainActivity.java

```diff
@@ -0,0 +1,148 @@
+     * When this method is called again, which happens whenever onCreate is called,
+     * it will return the pre-existing ViewModel associated with the specific Court-Counter JWMainActivity.
+     * This is what preserves the data.
+     * <p>
+     * {@link - https://developer.android.com/training/basics/fragments/communicating}
+     * {@link - https://medium.com/androiddevelopers/viewmodels-a-simple-example-ed5ac416317e}
+     *
+     * @see androidx.lifecycle.ViewModel
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        // Create a ViewModel the first time the system calls an activity's onCreate() method.
+        // Re-created activities receive the same MyViewModel instance created by the first activity.
+        ViewModelProviders
+                .of(this)
+                .get(JWViewModel.class);
+
+        // Bind all of my views to this activity
+        // TODO: I might get rid of bind because I'm not sure if Butterknife will stay up to date
+        ButterKnife.bind(this);
+
+        // Call Support Fragment's Manager to manage my fragments
+        FragmentManager fragmentManager = getSupportFragmentManager();
+
+        // Set OnNavigationItemListener
+        navigation.setOnNavigationItemSelectedListener(menuItem -> {
+
+            switch (menuItem.getItemId()) {
+                case R.id.navigation_jwplayer:
+                    Log.i("HYUNJOO", "JWMainActivity - navigation bar - jwplayerview");
+                    mCurrFragmentView = new JWPlayerViewExample();
+                    mCurrFragmentView.setArguments(bundle);
+                    break;
+                case R.id.navigation_settings:
+                    Log.i("HYUNJOO", "JWMainActivity - navigation bar - settings");
+                    mCurrFragmentView = new JWPlayerViewSettings();
+                    break;
+            }
+
+            // Replace my current Fragment view (container.getId) with mCurrFragmentView
+            fragmentManager
+                    .beginTransaction()
+                    .replace(mFrameLayout.getId(), mCurrFragmentView)
+                    .commit();
+
+            return true;
+
+        });
+
+        // My default nagivation page is JWPlayerView
+        navigation.setSelectedItemId(R.id.navigation_jwplayer);
+    }
+
+    /**
+     * Exit fullscreen when the user pressed the Back button
+     * Credits to: {@link - https://stackoverflow.com/questions/12210906/fragment-activity-catch-onkeydown-and-use-in-fragment#answer-12211422
+     */
+    @Override
+    public boolean onKeyDown(int keyCode, KeyEvent event) {
+        if (keyCode == KeyEvent.KEYCODE_BACK) {
+            if (mCurrFragmentView != null && mCurrFragmentView.getClass().getName().equalsIgnoreCase("com.example.jwplayerdemo.JWPlayerViewExample")) {
```

Comment:
Use `instanceof` here, it should be faster, e.g. `mCurrFragmentView instanceof JWPlayerViewExample`